readconf.c

```diff
@@ -1734,18 +1738,140 @@ read_config_file_depth(const char *filename, struct passwd *pw,
 
 	if ((f = fopen(filename, "r")) == NULL)
 		return 0;
+	if (flags & SSHCONF_CHECKPERM) {		
+#if WINDOWS
```

Comment:
To reuse the code in other places (on future need basis) we can move good amount of code to posix library.
1) IsFileOwnedByCurrentProcessId(char *file) 
2) IsFileWrittable(char *file, int fileOwners)
    int fileOwners will take the multiple values of this enum
        enum AccountOwners {
         CURRENT_USER,
         SYSTEM,
         LOCAL ADMIN,
         OTHERS
        }
 
This will reduce this huge chunk into few line of code, is much easy for others to understand and easy to reuse the code in future.